src/app/components/card/card.component.ts

```diff
@@ -36,48 +35,61 @@ export interface CardAction {
     action?: Action;
 }
 
+export interface CardRights {
+    name: string;
+    in: boolean;
+    right: 'reader' | 'writer';
+}
+
 @Component({
     selector: 'arlas-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.scss']
 })
 export class CardComponent implements AfterViewInit, OnInit {
 
-    @ViewChild('configMenu', { static: false }) public configMenu: ConfigMenuComponent;
+    @ViewChild('configMenu', {static: false}) public configMenu: ConfigMenuComponent;
     @Input() public card: Card;
     @Input() public userGroups: string[] = [];
     @Input() public publicOrg = false;
     @Output() public actionOnCard: Subject<CardAction> = new Subject<CardAction>();
 
     public action = Action;
-    public collectionColor = '#000000';
     public status = 'private';
     public NO_ORGANISATION = NO_ORGANISATION;
 
-    public readers: Array<{ name: string; in: boolean; }> = [];
-    public writers: Array<{ name: string; in: boolean; }> = [];
+    public rights: Array<CardRights> = [];
 
-    public constructor(
-        private colorService: ArlasColorService
-    ) {
+    public constructor() {
     }
+
     public ngOnInit(): void {
-        if (!!this.card) {
-            this.readers = this.card.readers.map(g => ({
-                name: g.name,
-                in: this.userGroups.indexOf(g.name) > -1
-            }));
-            this.writers = this.card.writers.map(g => ({
+        if (Boolean(this.card)) {
+            this.initDashboardWright();
+            this.initDashboardVisibility();
+        }
+    }
+
+    public initDashboardWright() {
+        const writers: CardRights[] = this.card.writers.map(g => ({
```

Comment:
I think the old implementation was missing a check if card.writers and card.readers exist and defined
could you please add this verification ?

Reply:
done